lib/facter/java_path.rb

```diff
@@ -0,0 +1,19 @@
+# Fact: java_path
+#
+# Purpose: get absolute path of java executable
+#
+# Resolution:
+#   Uses `readlink` to resolve the path of `/usr/bin/java`
+#
+# Caveats:
+#   Requires readlink
+#
+# Notes:
+#   None
+Facter.add(:java_path) do
+  setcode do
+    if Facter::Util::Resolution.which('readlink')
+      Facter::Util::Resolution.exec('readlink -e /usr/bin/java').strip
```

Comment:
... or Windows. Please just add the confine to `:kernel => "Linux"` if you do not want to support other platforms (which is totally fine with me!)